src/app/src/test/kotlin/org/vaccineimpact/api/tests/app_start/AllowedOriginFilterTests.kt

```diff
@@ -0,0 +1,97 @@
+package org.vaccineimpact.api.tests.app_start
+
+import com.nhaarman.mockito_kotlin.doReturn
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.verifyZeroInteractions
+import org.junit.Test
+import org.vaccineimpact.api.app.app_start.AllowedOriginsFilter
+import org.vaccineimpact.api.test_helpers.MontaguTests
+import spark.Request
+import spark.Response
+import javax.servlet.http.HttpServletResponse
+
+class AllowedOriginFilterTests : MontaguTests()
+{
+
+    @Test
+    fun `allows UAT`()
+    {
+        val sut = AllowedOriginsFilter(false)
+        val mockResponse = sut.handleMockRequest("https://support.montagu.dide.ic.ac.uk:10443/admin")
+        verify(mockResponse)
+                .addHeader("Access-Control-Allow-Origin", "https://support.montagu.dide.ic.ac.uk:10443/admin")
+    }
+
+    @Test
+    fun `allows science`()
+    {
+        val sut = AllowedOriginsFilter(false)
+        val mockResponse = sut.handleMockRequest("https://support.montagu.dide.ic.ac.uk:11443/admin")
+        verify(mockResponse)
+                .addHeader("Access-Control-Allow-Origin", "https://support.montagu.dide.ic.ac.uk:11443/admin")
+    }
+
+    @Test
+    fun `allows production`()
+    {
+        val sut = AllowedOriginsFilter(false)
+        val mockResponse = sut.handleMockRequest("https://montagu.vaccineimpact.org/admin")
+        verify(mockResponse)
+                .addHeader("Access-Control-Allow-Origin", "https://montagu.vaccineimpact.org/admin")
+    }
+
+    @Test
+    fun `does not allow localhost`()
+    {
+        val sut = AllowedOriginsFilter(false)
+        val mockResponse = sut.handleMockRequest("http://localhost")
+        verifyZeroInteractions(mockResponse)
+    }
+
+    @Test
+    fun `does not allow localhost on https`()
+    {
+        val sut = AllowedOriginsFilter(false)
+        val mockResponse = sut.handleMockRequest("https://localhost")
+        verifyZeroInteractions(mockResponse)
+    }
+
+    @Test
+    fun `does not allow randoms`()
+    {
+        val sut = AllowedOriginsFilter(false)
+        val mockResponse = sut.handleMockRequest("https://google.com")
+        verifyZeroInteractions(mockResponse)
+    }
+    @Test
+    fun `allows localhost`()
```

Comment:
Feel like this test should have a more descriptive name, or is just seems to contradict the earlier test `does not allow localhost`